prql-compiler/src/tests/test.rs

```diff
@@ -1507,6 +1507,7 @@ fn test_distinct_on() {
     FROM
       employees
     ORDER BY
+      department,
```

Comment:
Thanks, I didn't realize this was required. On reflection I should have added an integration test given I'm less familiar with this construct. (And I think that's a very reasonable thing for us to ask each other to do, so feel free to prod me on that sort of thing)

Reply:
Yes, I agree - let's be more diligent here.

We should also be converting the units tests we have into integration tests. Last few times that I did that it unearthed new bugs.

Reply:
Yes great.

(i do think the unit tests are still useful for the quick iterations — it's much easier to debug why bad SQL is being produced than why we're getting bad query results — but agree that there's plenty of room to expand the integration tests)